addpeer.go

```diff
@@ -96,17 +97,15 @@ func (p *AddPeerProtocol) AddPeer(ctx context.Context, peerAddr string) error {
 		return err
 	}
 	p.node.Peerstore().AddAddr(peerid, targetAddr, pstore.PermanentAddrTTL)
-	// create message data
-	req := &pbmsg.AddPeerRequest{
-		Message: fmt.Sprintf("AddPeer from %s", p.node.ID()),
-	}
 
-	s, err := p.node.NewStream(ctx, peerid, addPeerRequest)
-	if err != nil {
-		logger.FinishWithErr(spanctx, fmt.Errorf("Failed to open a new stream with peer %v, err: %v", peerid, err))
+	if err := p.node.Connect(ctx, p.node.Peerstore().PeerInfo(peerid)); err != nil {
```

Comment:
If we do replace `NewStream` with `Connect`, does that mean we can remove `AddPeerProtocol` entirely?

Reply:
Maybe for now? If in the future there are information to be exchanged when adding a new peer, `AddPeerProtocol` could be added back.

Reply:
Yes, I think for now we can remove `AddPeerProtocol` entirely, and change the function `AddPeer` to simply calls `Connect`